src/util/unesc.js

```diff
@@ -1,19 +1,96 @@
-const whitespace = '[\\x20\\t\\r\\n\\f]';
-const unescapeRegExp = new RegExp('\\\\([\\da-f]{1,6}' + whitespace + '?|(' + whitespace + ')|.)', 'ig');
+// Many thanks for this post which made this migration much easier.
+// https://mathiasbynens.be/notes/css-escapes
+
+const CSS_SPECIAL_CHARS = new Set([
+    "!", "\"", "#", "$", "%", "&", "'", "(", ")", "*", "+", ",", "-", ".", "/", ";", "<", "=", ">", "?", 
+    "@", "[", "]", "^", "`", "{", "|", "}", "~", "_", ":",
+]);
+
+const CSS_SPECIAL_CHARS_HEX_LONG = {}; 
+const CSS_SPECIAL_CHARS_HEX_SHORT = {};
+
+for (const char of Array.from(CSS_SPECIAL_CHARS)) {
```

Comment:
Sets are iterable, so we can even avoid casting to array here:

```suggestion
for (const char of CSS_SPECIAL_CHARS) {
```